### PR TITLE
(IAC-966) - MODULES-10522: Add support for the --condition parameter

### PR DESCRIPTION
**Pull request by adrianiurca**
Closes: #891

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -474,6 +474,20 @@ To prevent this issue, do not use MCollective to kick off Puppet runs. Use any o
...
+For ubuntu distributions `xtables-addons-common` package can be installed by running command: `apt-get install xtables-addons-common` or
+running a manifest:
+
+```puppet
+package { 'xtables-addons-common':
+  ensure => 'latest',
+}
+```
+
+For other distributions(RedHat, Debian, Centos etc) is required a manual installation of `xtables-addons` package.
```
Comment (sanfrancrisko) [inline README.md:489]:
```suggestion
For other distributions (RedHat, Debian, Centos etc) manual installation of the `xtables-addons` package is required.
```

File: spec/spec_helper_acceptance_local.rb [hunk 2]
```diff
@@ -44,15 +44,15 @@ def update_profile_file
 end
 
 RSpec.configure do |c|
+  # This flag is disabling some tests on docker/vagrant containers
+  # To enable tests on abs/vmpooler machines just set to `true` this flag
+  c.filter_run_excluding ubuntu_vmpooler: false
```
Comment (sanfrancrisko) [inline spec/spec_helper_acceptance_local.rb:49]: I'm not entirely sure I understand this - the comment states that the flag disables tests on Docker/Vagrant, but is then controlling vmpooler tests?

File: lib/puppet/provider/firewall/iptables.rb [hunk 3]
```diff
@@ -452,6 +455,8 @@ def self.rule_to_hash(line, table, counter)
     values = values.gsub(%r{(!\s+)?--tcp-flags (\S*) (\S*)}, '--tcp-flags "\1\2 \3"')
     # --hex-string output is in quotes, need to move ! inside quotes
     values = values.gsub(%r{(!\s+)?--hex-string "(\S*?)"}, '--hex-string "\1\2"')
+    # --condition output is in quotes, need to move ! inside quotes
+    values = values.gsub(%r{(!\s+)?--condition "(\S*?)"}, '--condition "\1\2"')
```
Comment (sanfrancrisko) [inline lib/puppet/provider/firewall/iptables.rb:459]: You could use `!` to mutate the value of `values`
```suggestion
    values.gsub!(%r{(!\s+)?--condition "(\S*?)"}, '--condition "\1\2"')
```

**Review by sanfrancrisko — CHANGES_REQUESTED** [review]
I would change the title and description of this PR as it is incorporating the work from https://github.com/puppetlabs/puppetlabs-firewall/pull/891 and adding tests to it. You could probably just take the title and description from that PR.

File: spec/spec_helper_acceptance_local.rb (hunk 2 above)
Comment (adrianiurca) [reply spec/spec_helper_acceptance_local.rb:49]: exactly

**Comment by mattock** [comment]
Thanks everyone for finishing what I started and failed to complete in time!

**Pull request merged**